File: app/src/main/java/uk/co/agfsoft/skyweather/presenter/ShowFavouritePresenter.java
package uk.co.agfsoft.skyweather.presenter;

import com.android.volley.VolleyError;

import javax.inject.Inject;

import uk.co.agfsoft.skyweather.model.Model;
import uk.co.agfsoft.skyweather.model.RawWeatherCity;
import uk.co.agfsoft.skyweather.view.ShowFavouriteView;
import uk.co.agfsoft.skyweather.volley.BaseVolleyRequest;



public class ShowFavouritePresenter {
    private final ShowFavouriteView showFavouriteView;
    private final Model model;
    private int cityId;

    @Inject
    public ShowFavouritePresenter(ShowFavouriteView showFavouriteView, Model model) {

        this.showFavouriteView = showFavouriteView;
        this.model = model;
    }

    public void viewReady() {
        if (cityId != 0) {
            submitWeatherRequest(cityId);
        }

    }

    private void submitWeatherRequest(int cityId) {
        model.getWeatherForCity(cityId, new BaseVolleyRequest.Listener<RawWeatherCity>() {
            @Override
            public void onSuccess(RawWeatherCity response) {
                showFavouriteView.updateWeatherDisplay(response);
            }

            @Override
            public void onError(VolleyError error) {
                showFavouriteView.showError();

            }
        });
    }


    public void setCityId(int cityId) {
        this.cityId = cityId;
    }


}

File: app/src/main/kotlin/uk/co/agfsoft/skyweather/presenter/ShowFavouritePresenter.kt
package uk.co.agfsoft.skyweather.presenter

import com.android.volley.VolleyError

import javax.inject.Inject

import model.Model
import model.RawWeatherCity
import uk.co.agfsoft.skyweather.view.ShowFavouriteView
import uk.co.agfsoft.skyweather.volley.BaseVolleyRequest


open class ShowFavouritePresenter @Inject
constructor(private val showFavouriteView: ShowFavouriteView, private val model: Model) {
    private var cityId: Int = 0

    fun viewReady() {
        if (cityId != 0) {
            submitWeatherRequest(cityId)
        }

    }

    private fun submitWeatherRequest(cityId: Int) {
        model.getWeatherForCity(cityId, object : BaseVolleyRequest.Listener<RawWeatherCity> {
            override fun onSuccess(response: RawWeatherCity) {
                showFavouriteView.updateWeatherDisplay(response)
            }

            override fun onError(error: VolleyError) {
                showFavouriteView.showError()

            }
        })
    }


    fun setCityId(cityId: Int) {
        this.cityId = cityId
    }


}
